### DISCUSSION TITLE
Do my extra java arguments optimize the game or are they just unnecessary arguments?

### DISCUSSION THREAD

**Discussion by Fleepe** [TheUsefulLists/UsefulMods#108] [Q&A]
I recently read the Java arguments optimization that you wrote and I didn't see some of the arguments I have. I don't know java as a programming language and don't know what each argument means - I just copy paste them and play.

Minecraft version: 1.12.2
Java: Eclipse Temurin JDK with Hotspot 8u332-b09 (x64)

Aikar's arguments: 
`-Xms10G -Xmx10G -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1`

My arguments that do not match or are not written by Aikar: 
`-XX:MaxGCPauseMillis=16 -XX:+UnlockDiagnosticVMOptions -XX:UseAVX=2 -XX:-UseBiasedLocking -XX:SoftRefLRUPolicyMSPerMB=2000 -XX:+EagerXrunInit -XX:+UseNewLongLShift -XX:+UseXmmI2D -XX:+UseXmmI2F -XX:+OptoScheduling -XX:+OptoBundling -XX:+UseStringDeduplication -XX:+UseFastUnorderedTimeStamps -XX:UseSSE=4 -XX:+UseAES -XX:+UseAESIntrinsics -Xverify:none -XX:TargetSurvivorRatio=50 -XX:+AggressiveOpts`

The main question is do I need to remove them or do something with them so that Minecraft has more fps? Maybe I need to add something to these arguments?

**Comment by lrmtheboss** [comment]
Most of those I've never seen before, the `-XX:MaxGCPauseMillis=16 -XX:+UnlockDiagnosticVMOptions` ones I have seen before but don't remember where. My suggestion would be to start with just Aikar's flags, play it awhile, add those others and play some more. If it's going to hurt or help your game it should be pretty noticeable fairly quickly

**Reply by Fleepe** [reply]
Well, I tested both and this is what I noticed: with Aikar's arguments I have 3-4 more fps, but with the arguments that I had when loading chunks, not the world, the game is much more stable. Perhaps we need to involve people who know how the arguments that I have work, maybe they will explain what needs to be removed or changed. Or just do more tests and find out if these arguments are needed or not.
